ang/civicase/Actions.js

```diff
@@ -1,386 +1,364 @@
-                query: {
-                  reset: 1,
-                  case_id: currentCase.id
-                }
-              };
-
-              return popupPath;
+            if (cases.length === 1) {
+              popupPath.query.caseid = cases[0].id;
             }
 
-          });
+            return popupPath;
+          },
+          printMerge: PrintMergeCaseAction.getPath,
+          exportCases: function (cases) {
+            var caseIds = _.collect(cases, 'id');
+            var popupPath = {
+              path: 'civicrm/export/standalone',
+              query: {
+                reset: 1,
+                entity: 'Case',
+                id: caseIds.join()
+              }
+            };
+            return popupPath;
+          },
+
+          linkCases: function (case1, case2) {
+            var activityTypes = CRM.civicase.activityTypes;
+            var link = {
+              path: 'civicrm/case/activity',
+              query: {
+                action: 'add',
+                reset: 1,
+                cid: case1.client[0].contact_id,
+                atype: _.findKey(activityTypes, {name: 'Link Cases'}),
+                caseid: case1.id
+              }
+            };
 
-          // Open popup if callback returns a path & query
-          if (result) {
-            // Add refresh data
-            if ($scope.popupParams) {
-              result.query.civicase_reload = $scope.popupParams();
+            if (case2) {
+              link.query.link_to_case_id = case2.id;
             }
 
-            // Mimic the behavior of CRM.popup()
-            var formData = false;
-            var dialog = CRM.loadForm(CRM.url(result.path, result.query))
-              // Listen for success events and buffer them so we only trigger once
-              .on('crmFormSuccess crmPopupFormSuccess', function (e, data) {
-                formData = data;
-              })
-              .on('dialogclose.crmPopup', function (e, data) {
-                if (formData) {
-                  element.trigger('crmPopupFormSuccess', [dialog, formData]);
-                }
+            return link;
+          },
+
+          print: function (selectedCase) {
+            var url = CRM.url('civicrm/case/report/print', {
+              all: 1,
+              redact: 0,
+              cid: selectedCase.client[0].contact_id,
+              asn: 'standard_timeline',
+              caseID: selectedCase.id
+            });
+            var win = window.open(url, '_blank');
+            win.focus();
+          },
+
+          gotoWebform: function (selectedCase, path, clientId) {
+            clientId = 'cid' + clientId;
+            var url = CRM.url(path, {
+              case1: selectedCase.id,
+              [clientId]: selectedCase.client[0].contact_id
+            });
+            var win = window.open(url, '_blank');
+            win.focus();
+          },
+
+          lockCases: function (currentCase) {
```

Comment:
Can't we directly return the array and add the info in the function docs.
All the functions lacks docs. Please add docs to each function

Reply:
This whole file was refactored to fix the linting issues, and also to apply our styleguide. But as I have not worked specifically on this file/functionality, I am not aware what each function exactly does, so I am skipping any kind of documentation/major refactoring.